auction-house/program/src/bid/mod.rs

```diff
@@ -123,6 +219,40 @@ pub fn private_bid<'info>(
     )
 }
 
+/// Create a private bid on a specific SPL token that is *held by a specific wallet*.
+pub fn private_bid_with_auctioneer<'info>(
+    ctx: Context<'_, '_, '_, 'info, BuyWithAuctioneer<'info>>,
+    trade_state_bump: u8,
+    escrow_payment_bump: u8,
+    buyer_price: u64,
+    token_size: u64,
+    _ah_auctioneer_pda_bump: u8,
+) -> ProgramResult {
+    auction_bid_logic(
+        ctx.accounts.wallet.to_owned(),
+        ctx.accounts.payment_account.to_owned(),
+        ctx.accounts.transfer_authority.to_owned(),
+        ctx.accounts.treasury_mint.to_owned(),
+        ctx.accounts.token_account.to_owned(),
+        ctx.accounts.metadata.to_owned(),
+        ctx.accounts.escrow_payment_account.to_owned(),
+        ctx.accounts.authority.to_owned(),
+        &mut ctx.accounts.auction_house,
```

Comment:
`accounts` is a `&mut T` so I can't move the auction_house value out of that, and I can't copy it because it's behind a `Box` (to avoid blowing the compute budget). 

This is the error if you remove the `&mut`: 

```
cannot move out of `ctx.accounts.auction_house` which is behind a mutable reference
move occurs because `ctx.accounts.auction_house` has type `Box<anchor_lang::prelude::Account<'_, state::AuctionHouse>>`, which does not implement the `Copy` traitrustc[E0507](https://doc.rust-lang.org/error-index.html#E0507)
```

Is there an issue with using a mutable reference here?